x/blob/client/cli/payforblob.go

```diff
@@ -57,10 +68,27 @@ func CmdPayForBlob() *cobra.Command {
 	}
 
 	flags.AddTxFlagsToCmd(cmd)
+	cmd.PersistentFlags().Uint8(FlagShareVersion, 0, "Specify the share version")
 
 	return cmd
 }
 
+func getShareVersion(shareVersion uint8) (uint8, error) {
+	// Check if share version is present in SupportedShareVersions
+	result := false
+	for _, val := range appconsts.SupportedShareVersions {
+		if val == shareVersion {
+			result = true
+			break
+		}
+	}
+
+	if result {
+		return shareVersion, nil
+	}
+	return uint8(0), fmt.Errorf("share version %d is not supported", shareVersion)
+}
+
```

Comment:
this may be simplified

```go
func validateShareVersion(shareVersion uint8) error {
	if !slices.Contains(appconsts.SupportedShareVersions, shareVersion) {
		return fmt.Errorf("share version %d is not supported", shareVersion)
	}
	return nil
}
```

Reply:
in a future refactor we can consider consolidating all the code that validates share versions into one exported function in the `shares` package

Reply:
Thanks, Updated